### PR TITLE
Traitors can purchase weapon cases for the Makarov and Riot Toy Pistol. Basic ammo comes in ammo packs. Other stuff.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Makarovs and Donksoft Riot Toy Pistols now come in weapon cases. These cases come with the gun, two spare magazines and a box of spare loose ammo.

Basic ammunition for these guns come in case packs of three, with a spare box of loose ammo. The basic ammo for Makarovs is slightly more expensive.

Donksoft Toy Pistols from the uplink (and given to clown ops) now deal substantially more damage. For riot darts, this goes from 25 to 35 stamina force. However, the case now costs 6 TC.

Makarov and Donksoft toy pistols have had their magazines increased by 4 bullets, for a total of 12. The Ansem is still at 8.

The pistol cases now have a unique sprite thanks to SmartKar.

image

All included cases come with a disposal bomb built into the case. Use Alt-Right-Click on the case to start the countdown, and stand back. Or chuck it at someone you don't like.

## Why It's Good For The Game

>Makarovs and Donksoft Riot Toy Pistols now come in weapon cases. These cases come with the gun, two spare magazines and a box of spare loose ammo.

>Basic ammunition for these guns come in case packs of three, with a spare box of loose ammo. The basic ammo for Makarovs is slightly more expensive.

A long time ago, I proposed to a maintainer that maybe traitor weapons should be bundled together with some starting ammo, to give them a little bit of a necessary boost for longevity for their cost. This being in a state of the game with only 20 TC and no way to get more. I thought progtot may alleviate this issue somewhat, but I don't believe it has at all. 

I narrowed this down to just the Makarov and Toy Pistol as I think they're the ones in need of help, and my reasoning is thus;

A) The TC value is extremely deceptive. 

The weapons themselves are not very good at doing anything without additionally putting in more TC to load up on ammunition or support equipment. While it does say 7 TC on the header for the Makarov, if you want it to be silent, you have to spend more TC on a suppressor (3 TC). Then, if you're not entirely sure that only 8+1 shots you get from the beginning is enough to take out a target (maybe they're known to have armor), then you may need to get either specialized ammo or additional ammunition. At a certain point, it begs the question 'why didn't I just buy the bigger stick and get more value out of my purchase as well as more reliability?' Particularly since drawing heat probably means all that TC is just going straight into more ammo, one way or another. Or a bigger stick if you're doing side objectives, making that early purchase redundant.

B) Whenever you look at either, I want you to consider. 'Could I get better results by just getting a station weapon?' 

The answer is almost universally 'yes'. Even some improvised weapons can be more reliable. Even spending TC on getting that weapon (like C-4) is miles more worthwhile than spending TC to get either weapon upfront.

C) It significantly overvalues autolathe access just to make the weapon feel less like a ripoff. If the uplink can't in of itself justify using the weapons, they're just flat out not good. and that's assuming players even know to keep magazines.

It's just a bad value purchase and deceptive in just how expensive it can end up to use them. The weapon is overshadowed by fairly comparable items once you factor additional expenditures. 

Even the toy pistol is a bad value purchase and it's literally just 2 TC, because the damage it does is dogshit. It's worse than a disabler. It's worse than a sleepy pen for value to effect. It's so shit, people put them on maps for free grabs. Let's resolve that with the following.

> Donksoft Toy Pistols from the uplink (and given to clown ops) now deal substantially more damage. For riot darts, this goes from 25 to 35 stamina force. However, the case now costs 6 TC.

Subtle weapon, good, reliable damage. Fantastic for kidnapping at 40 damage. All without ever actually inflicting a real point of damage. Great for clowns, great for pacifist tots. Decent deniability. Silicons may have a harder time justifying an intervention. Genuinely a budget tool that increases the value of any TC you put into it. Since it has so much more value, this is why I've increased the price to offset this. It is actually competing with similar tools well enough to not go too overboard.

> Makarov and Donksoft toy pistols have had their magazines increased by 4 bullets, for a total of 12. The Ansem is still at 8.

Longevity is the only thing Makarovs seem to want to claim to have over the revolver, and it isn't much more longevity from just the gun itself (7 bullets on the revolver compared to 8+1 on the pistol). If we're talking real longevity, and particularly if you're considering getting a silencer, than the ebow is literally 3 TC more expensive, silent, and has endless ammunition with high damage output. 

You still need to dump a lot of TC into the Makarov to satiate its ammo needs if you happen to be skirmishing a lot. By comparison, just getting a single laser can often times do a much better job at skirmishing than the tot with a Makarov can. And do comparable damage no less. From experience, I've always done better with lasers than the Makarov.

Making it more directly able to maintain a good ammo count during a fight hopefully makes the Makarov feel more like a value purchase for what I think should be its strength. Staying power and the ability to be aggressive with ammo expenditure. Particularly against larger numbers, which traitors are almost always expected to go up against. If those opponents have lasers, the Makarov just always gets outgunned by an absurd amount (lasers on their own have like 16 shots, so you can do the math if there are even two people with a laser each).

Simply put; let's not make make tots feel like a dumbass for not just getting a laser themselves or buying a ebow to maintain firepower over long fights, especially with rechargers usually being in pretty nicely secluded locations for them to access.

## Changelog

:cl: NecromancerAnne (code), SmArtKar (sprites)
balance: Makarovs and Toy Pistols come in weapon cases. Complete with spare ammo.
balance: Basic ammo for either weapon comes in weapon cases of three extra magazines at an affordable price.
balance: Donksoft Toy Pistols from the uplink are much stronger than their standard counterparts, but now priced at 6 TC.
balance: Makarovs and Toy pistols have a magazine capacity of 12 rounds.
balance: Gun/Ammo cases from the traitor uplink can be destroyed by activating the disposal bomb. Press Alt-Right-Click on the case to start the timer.
/:cl:

### CODE REVIEW

**Comment by Sadboysuss** [comment]
anything to see more hitman style loadouts with silenced pistols

**Comment by carlarctg** [comment]
weapon cases are pretty annoying to dispose of because they can't be folded up unlike boxes

**Comment by SmArtKar** [comment]
> weapon cases are pretty annoying to dispose of because they can't be folded up unlike boxes

clearly they should be shipped with an explosive charge so they could also be used as a low power grenades in a pinch

**Comment by mc-oofert** [comment]
> > weapon cases are pretty annoying to dispose of because they can't be folded up unlike boxes
> 
> clearly they should be shipped with an explosive charge so they could also be used as a low power grenades in a pinch

facts

**Comment by OuroborusReporter** [comment]
Smar**k**Kar (sprites)

**Comment by SimplySmiley123** [comment]
> > weapon cases are pretty annoying to dispose of because they can't be folded up unlike boxes
> 
> clearly they should be shipped with an explosive charge so they could also be used as a low power grenades in a pinch

Honestly that would be extremely funny, even funnier if it could be used as a trap against investigating seccies / detectives, leave the evidence, detonate it, earn an advantage over whoever was investigating in the moment.

**Comment by MoonDeer728** [comment]
нельзя так делать, оставь в покое аплинк

**Comment by Loafin34** [comment]
the toy pistol should have a more colorful design for the case. also add the cases to donksoft please...

File: code/modules/projectiles/guns/ballistic/toy.dm [hunk 1]
```diff
@@ -31,6 +32,9 @@
 	magazine = new /obj/item/ammo_box/magazine/toy/pistol/riot(src)
 	return ..()
 
+/obj/item/gun/ballistic/automatic/pistol/toy/riot/clandestine
+	projectile_damage_multiplier = 1.6
```
Comment (SmArtKar) [inline code/modules/projectiles/guns/ballistic/toy.dm:36]: 40 is a very important threshold for stamloss, as 40/75 are big stamina breakpoints. 35 will still stamcrit in 3 shots but won't apply slowdown after the first/increase slowdown after the second shot.
```suggestion
/obj/item/gun/ballistic/automatic/pistol/toy/riot/clandestine
	projectile_damage_multiplier = 1.4
```

**Review by SmArtKar — DISMISSED** [review]

**Comment by vendingmachine685** [comment]
give traitors nukie weapons

**Comment by SomeoneYouProbablyKnow** [comment]
> Honestly that would be extremely funny, even funnier if it could be used as a trap against investigating seccies / detectives, leave the evidence, detonate it, earn an advantage over whoever was investigating in the moment.

If only, but det scanner is ranged and we have quite a few super easy ways not to be gotten by explosives likes that but still very amusing. 

As per the thing itself... oh no a box that's going to brand me as a traitor, brother you are carrying a gun. You (just like literally 99% of traitors) are fucked in a search, so shoot them. If you wanted to mess with sec then dont get ANY items.The uplink doesnt have search proof items. Sad as that is that's the state of things.

Also hell yea. I've used the makarov plenty but  it's a whole ordeal to print more ammo in advance get clips etc etc. This just simplifies the whole process and gives traitors a cheap self defence gun to put em on even grounds with one (1) security officer.

File: code/modules/projectiles/guns/ballistic/toy.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/projectiles/guns/ballistic/toy.dm:36]: Fair enough.

**Comment by Loafin34** [comment]
storage implant in question:

**Comment by mc-oofert** [comment]
bro clearly never hid loot somewhere and let themselves be searched

**Comment by necromanceranne** [comment]
You can now dispose of your case using Alt-Right-Click. 

Just don't stand too close to it when it explodes.

**Comment by DrAmazing343** [comment]
holy FUCKING PEAK PR IT EXPLODES NOW???

**Comment by MrMelbert** [comment]
I don't really see the argument that the ebow is comparable to the makarov, takes like 15 seconds to crit someone with an ebow vs like 2 seconds with a makarov

**Comment by SomeoneYouProbablyKnow** [comment]
> storage implant in question:

That's gambling on security not having a stick up their ass and trying a surgery on you. I've seen sec find storage implants on the most random people who should not have been tested for it. 

> bro clearly never hid loot somewhere and let themselves be searched

This is much more reliable than storage implant or any of our stealthy items yea. That or just not buying anything, leaving sec confused untill the moment you need to go full hog. But there are no items in the uplink that help with being searched. A box or a gun and a box makes little difference.

It exploding now is sick tho

**Comment by SimplySmiley123** [comment]
> I don't really see the argument that the ebow is comparable to the makarov, takes like 15 seconds to crit someone with an ebow vs like 2 seconds with a makarov

They are as good as dead if you hit someone with an e-bow since they get glued to the floor after 2 shots, with zero ammo issues on it since it's infinite ammo, self-charging weapon, alongside not requiring you to buy a silencer separately to not announce to everyone around your location "someone's getting murdered here".

**Comment by MrMelbert** [comment]
> They are as good as dead if you hit someone with an e-bow since they get glued to the floor after 2 shots, with zero ammo issues on it since it's infinite ammo, self-charging weapon, alongside not requiring you to buy a silencer separately to not announce to everyone around your location "someone's getting murdered here".

But the difference is you can still call for help over comms over the next 15 seconds of them tickling you to death, when a makarov doesn't afford you the same grace

**Comment by necromanceranne** [comment]
There are tools in the uplink to solve radio callouts. You also can just, like, stab them to death with literally any other weapon. Even kicking them to death doesn't take too terribly long. I'm more concerned with longevity to TC though.

**Comment by SomeoneYouProbablyKnow** [comment]
The ebow and makarov are different tools and occupy different niches buy there is some overlap. Mainly as a semi-commital side-arm and that's exactly the aspect of the gun that's being buffed here.
You won't be going on a killing spree and you probably won't be buying it to off an armed target unless you're really hard on cash. And even then if the goal's a single assassination you probably dont care about the extra ammo.

**Comment by SimplySmiley123** [comment]
> > They are as good as dead if you hit someone with an e-bow since they get glued to the floor after 2 shots, with zero ammo issues on it since it's infinite ammo, self-charging weapon, alongside not requiring you to buy a silencer separately to not announce to everyone around your location "someone's getting murdered here".
> 
> But the difference is you can still call for help over comms over the next 15 seconds of them tickling you to death, when a makarov doesn't afford you the same grace

Or you murder them with a melee weapon of choice, or you use any sort of radio jamming / EMP to render their radio useless.
Same as makarov really, you're not gonna kill someone off before they call for any help if they aren't AFK if you're _just_ using your singular murder tool and absolutely nothing else.

**Comment by mc-oofert** [comment]
it is VERY easy to silence someone by shooting them in the face while theyre down before they get a chance to react

**Comment by MrMelbert** [comment]
> Or you murder them with a melee weapon of choice, or you use any sort of radio jamming / EMP to render their radio useless. Same as makarov really, you're not gonna kill someone off before they call for any help if they aren't AFK if you're _just_ using your singular murder tool and absolutely nothing else.

I think you underestimate how quick the pistol crits (I mean it's 3-4 shots). It's often faster than most people can react (or at the bare minimum get out `;HELPMAINT`). Especially when it's suppressed it often catches people by surprise 

And if we're saying "just buy another weapon lol" or "just buy a jammer lol" then the argument about "why buy a makaraov + suppressor when you could buy an ebow" falls apart

**Comment by EnterTheJake** [comment]
> > Or you murder them with a melee weapon of choice, or you use any sort of radio jamming / EMP to render their radio useless. Same as makarov really, you're not gonna kill someone off before they call for any help if they aren't AFK if you're _just_ using your singular murder tool and absolutely nothing else.
> 
> I think you underestimate how quick the pistol crits (I mean it's 3-4 shots). It's often faster than most people can react (or at the bare minimum get out `;HELPMAINT`). Especially when it's suppressed it often catches people by surprise
> 
> And if we're saying "just buy another weapon lol" or "just buy a jammer lol" then the argument about "why buy a makaraov + suppressor when you could buy an ebow" falls apart

The makarov is still a bit too niche for its pricepoint.

 it's good at killing crewmates but falls pretty hard againts security, while the ebow doesn't.

**Comment by Loafin34** [comment]
having shot the makarov at people while using stormtrooper aim, people are too busy trying to dodge to yell out ;HELP BAR without getting critted.

**Comment by necromanceranne** [comment]
> I think you underestimate how quick the pistol crits (I mean it's 3-4 shots). It's often faster than most people can react (or at the bare minimum get out `;HELPMAINT`). Especially when it's suppressed it often catches people by surprise
> 
> And if we're saying "just buy another weapon lol" or "just buy a jammer lol" then the argument about "why buy a makaraov + suppressor when you could buy an ebow" falls apart

I think you're picking and choosing what arguments to address as opposed to my argument as a whole and how exactly I'm suggesting the ebow is used. My point with the ebow has to do with longevity and particularly, continued relevance even in a situation I would say things are falling apart. You can apply it in an assassination, I've literally done it time and time again. It's just a different approach. But sure, walking up to an AFK player and shooting them in the side of the head is faster and they may not even get a callout (but did you deal with sensors? What about an alert AI? What if they're an augmented human?). But that is the _the absolute ideal scenario_. Where nothing has gone wrong and there are no external means by which your assassination could be foiled. 

Could you do that to the captain? Could you do that to the HoS? What about if your target is somehow prepared for a fight and isn't just some dude sitting at a xenobio console, doing zen gardening?

Since we can apply the ebow in these scenarios and be successful, but also additional scenarios, such as engaging opponents and in self-defense/escapes, since we likely will be drawing heat if we do engage these opponents, we are prepared for the unknown and for the worse case scenario too. 

This is something that newer players do not operate well in and rarely prepare for. And thus, my suggestion for additional ammo does the same for the Makarov. Help with the worst case scenario and unforeseen circumstances to give them a fighting chance before they had even thought to consider it. Maintain the strength you see in it, but also cover the weakness that keeps more experienced players from wanting to use it. Hopefully, by being given some ammo for backup, the new player doesn't feel like they were cheated because they did not realize that just the gun itself wasn't going to be enough when things went poorly, because they, and seemingly even you, don't account for worst case scenarios. Most new players haven't developed the skills yet to be accounting for that, and they're my primary focus in every change I make.

**Comment by Loafin34** [comment]
Why did the makarov and toy pistol get 4 extra ammo? that makes it 88 bullets for 9 tc. (makarov case = 44, assuming it starts off unloaded, and the 9mm box is 20,). The value is a bit too great.

**Review by Ghommie — APPROVED** [review]
let's see how it goes, the guncase disposing looks good on paper.

**Pull request merged**